examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -277,93 +299,141 @@ def unflatten(theta: array) -> Dict[Handler, array]:
-        blown_up = _blew_up(results_this_step['complex']) or _blew_up(results_this_step['solvent'])
-
-        # note: unflatten_grad and unflatten_theta have identical definitions for now
-        flat_loss_grad, unflatten_grad = flatten(loss_grads)
-        flat_theta, unflatten_theta = flatten(ordered_params)
-
-        # based on current estimate of (loss, grad, and simulation stability), return a conservative step to take in parameter space
-        theta_increment = truncated_step(flat_theta, loss, flat_loss_grad,
-                                         step_lower_bound=_compute_step_lower_bound(loss, blown_up))
-        param_increments = unflatten_theta(theta_increment)
-
-        # for any parameter handler types being updated, update in place
-        for handle in ordered_handles:
-            handle_type = type(handle)
-            if handle_type in param_increments:
-                print(f'updating {handle_type.__name__}')
-
-                # TODO: careful -- this must be a "+=" or "-=" not an "="!
-                handle.params += param_increments[handle_type]
-
-                increment = param_increments[handle_type]
-                increment = increment[increment != 0]
-                min_update = 0.0
-                max_update = 0.0
-                if len(increment):
-                    min_update = np.min(increment)
-                    max_update = np.max(increment)
-                # TODO: replace with a function that knows what to report about each handle type
-                print(
-                    f'updated {len(increment)} params by between {min_update:.4f} and {max_update:.4f}')
-
-        t1 = time()
-        elapsed = t1 - t0
-
-        print(f'completed forcefield-updating step {step} in {elapsed:.3f} s !')
-
-        # save du_dls snapshot
-        path_to_du_dls = output_path.joinpath(f'du_dls_snapshot_{step}.npz')
-        print(f'saving du_dl trajs to {path_to_du_dls}')
-        du_dls_dict = dict()  # keywords here must be strings
-        for stage, results in results_this_step.items():
-            du_dls_dict[stage] = _results_to_arrays(results)[1]
-        np.savez(path_to_du_dls, **du_dls_dict)
-
-        # also save information about this step's parameter gradient and parameter update
-        # results to npz
-        path_to_npz = output_path.joinpath(f'theta_grad_loss_snapshot_{step}.npz')
-        print(f'saving theta, grad, loss snapshot to {path_to_npz}')
-        np.savez(
-            path_to_npz,
-            theta=np.array(flat_theta),
-            grad=np.array(flat_loss_grad),
-            loss=loss
-        )
+    step_inds = []
+    for epoch in range(num_epochs):
+        inds = np.arange(len(training.data))
+        np.random.shuffle(inds)
+        step_inds.append(inds)
 
-        # TODO: same for xs and du_dps snapshots
+    np.save(output_path.joinpath('step_indices.npy'), np.hstack(step_inds)[:args.param_updates])
```

Comment:
step indices is now only the training steps, with the validation set never shifting order (correct me if I am wrong in thinking we don't need to shuffle the validation set)

Reply:
This sounds right